sisl/geom/category/_coord.py

```diff
@@ -232,3 +233,55 @@ def __eq__(self, other):
                         return False
                 return True
         return False
+
+# The following code is just to make the use of coordinate categories more convenient
+# We will create one individual category for each direction. In this way, one can access
+# them directly in the keyword category builder. E.g.: `AtomCategory(x=(1,2))` creates `AtomXYZ(x=(1,2))`
+# without needing to nest "x", as in `AtomCategory(xyz={"x": (1,2)})`.
+
+
+class CoordMeta(CategoryMeta):
+    """
+    Metaclass to give extra functionality to individual coordinates categories.
+
+    The classes generated by this metaclass can use `<` and `>` to create categories.
+    See examples.
+
+    Examples
+    ----------
+
+    If `AtomX` has been built using this metaclass:
+    >>> AtomX < 5 == AtomX((None, 5)) == AtomXYZ(x=(None, 5))
+    >>> AtomX > 5 == AtomX((5, None)) == AtomXYZ(x=(5, None))
+    """
+
+    def __lt__(self, other):
+        return self(None, other)
+
+    def __gt__(self, other):
+        return self(other, None)
```

Comment:
also `__le__` and `__ge__`, in fact I think this is what is being done internally anyways

Reply:
Also, would `AtomX in (-2, 3)` be counter-intuitive? Would users assume X _has_ to be `-2` or `3`?

Reply:
Hmm good point. 

> Would users assume X has to be -2 or 3?

I think this would be the logical thing to believe knowing what `in` does in python

Since `AtomX(-2,3)` already achieves the goal, maybe it is unnecessary. But I'll add it if you want. This is the reason I have also not bothered to support ` -2 < AtomX < 3`.

Reply:
Well, the problem with `in` is that `2 in [1, 3]` will be false, that was my concern... Ok, lets keep it out for now!

Reply:
Yes, I was agreeing with that here :)

> I think this would be the logical thing to believe knowing what `in` does in python

Reply:
> also __le__ and __ge__, in fact I think this is what is being done internally anyways

We should probably differentiate somehow between `gt` and `ge`, no?

Reply:
> Yes, I was agreeing with that here :)
> 
> > I think this would be the logical thing to believe knowing what `in` does in python

Ok, it was just that that is not the python interpretation. But I think we agree that this *could* be useful, lets see if anybody wants it.


> We should probably differentiate somehow between `gt` and `ge`, no?

I agree. It would just require that `<direction>_gt` gets passed correctly and with expanded keywords.

Reply:
I have tried to play with `__contains__` I can't get it to work. I will push to this PR some changes.

Reply:
> Ok, it was just that that is not the python interpretation.

I'm getting confused now, the python interpretation is that `2 in [1, 3]` is `False`, right? I was agreeing with this and therefore saying that it is probably not worth it to implement `AtomX in [1, 3]` for ranges, specially since one can already do `AtomX(1,3)`. Or do you mean that you want `AtomX in [1, 3]` to actually match only 1 and 3?

Reply:
> > Ok, it was just that that is not the python interpretation.
> 
> I'm getting confused now, the python interpretation is that `2 in [1, 3]` is `False`, right? I was agreeing with this and therefore saying that it is probably not worth it to implement `AtomX in [1, 3]` for ranges, specially since one can already do `AtomX(1,3)`. Or do you mean that you want `AtomX in [1, 3]` to actually match only 1 and 3?

Ok, then we agree. :+1: